arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/Result.kt

```diff
@@ -0,0 +1,211 @@
+package arrow.core
+
+import kotlin.Result.Companion.failure
+import kotlin.Result.Companion.success
+
+@PublishedApi
+internal inline val unitResult: Result<Unit>
+  inline get() = success(Unit)
+
+public inline fun <A, B> Result<A>.flatMap(transform: (value: A) -> Result<B>): Result<B> =
+  fold({ a ->
+    try {
+      transform(a)
+    } catch (e: Throwable) {
+      failure(e)
+    }
+  }) { throwable ->
+    failure(throwable)
+  }
```

Comment:
Well, I guess `mapCatching { transform(it).getOrThrow() }` is actually a better implementation here since that preserves all the catching capabilities which were not covered here in the `failure` case.